src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,57 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserInvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_LENGTH = 6;
+    public static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        dataVerification(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void dataVerification(User user) {
+        if (user == null) {
+            throw new UserInvalidDataException(
+                    "You cannot pass a null value to a method");
+        }
+        if (user.getLogin() == null) {
+            throw new UserInvalidDataException(
+                    "User login cannot be null");
+        }
+        if (user.getPassword() == null) {
+            throw new UserInvalidDataException(
+                    "User password cannot be null");
+        }
+        if (user.getAge() <= 0) {
+            throw new UserInvalidDataException(
+                    "User age cannot be less than 0 or equal to 0. "
+                            + "Min allowed age is " + MIN_AGE);
```

Comment:
1. 0 in condition doesn't seem valid
2. Please use your constants in error message too in order to change value in one place in case you require to change logic for another value
```suggestion
        if (user.getAge() <= MIN_AGE) {
            throw new UserInvalidDataException(
                    "User age cannot be less than " + MIN_AGE
                            + ". Min allowed age is " + MIN_AGE);
```